syntax = "proto3";
package bastionlab;

message Empty {}

message ChallengeResponse {
    bytes value = 1;
}

message SessionInfo {
    bytes token = 1;
<<<<<<< HEAD
    uint64 expiry_time = 2;

=======
    // This is the time in miliseconds before the token will expire.
    uint64 expiry_time = 2;
>>>>>>> 0b64948 (Refresh session before request)
}

message ClientInfo {
    string uid = 1;
    string platform_name = 2;
    string platform_arch = 3;
    string platform_version = 4;
    string platform_release = 5;
    string user_agent = 6;
    string user_agent_version = 7;
    bool is_colab = 8;
}

service SessionService {
    rpc GetChallenge (Empty) returns (ChallengeResponse) {}
    rpc CreateSession (ClientInfo) returns (SessionInfo) {}
}
